### PR TITLE
Adds /image/proc/copy_overlays

### PR DESCRIPTION
**Pull request by Cyberboss**
Needed

### CODE REVIEW

File: code/controllers/subsystem/processing/overlays.dm [hunk 1]
```diff
@@ -183,10 +183,25 @@ PROCESSING_SUBSYSTEM_DEF(overlays)
 
 //TODO: Better solution for these?
 /image/proc/add_overlay(x)
-	overlays += x
+	overlays |= x
```
Comment (RemieRichards) [inline code/controllers/subsystem/processing/overlays.dm]: why.
```|=``` is ungodly slow in comparison to ```+=```

File: code/controllers/subsystem/processing/overlays.dm [hunk 2]
```diff
@@ -183,10 +183,25 @@ PROCESSING_SUBSYSTEM_DEF(overlays)
...
+
+/image/proc/copy_overlays(atom/other, cut_old)
+	if(!other)
+		if(cut_old)
+			cut_overlays()
+		return
+
+	var/list/cached_other = other.our_overlays
+	if(cached_other)
+		if(cut_old || overlays.len)
```
Comment (RemieRichards) [inline code/controllers/subsystem/processing/overlays.dm]: ```!overlays.len```, surely? to match the other proc.

**Comment by RemieRichards** [comment]
Issue:
Atoms can copy overlays from Atoms
Images can copy overlays from Atoms
Atoms can't copy overlays from Images (no using mutable appearances as temporary overlay holders)
Images can't copy overlays from Images (same as above but a rarer use-case)

**Comment by Cyberboss** [comment]
I'd rather not add istype overhead to these procs. For the last two cases `add_overlay(image.overlays)` can be used)

**Comment by RemieRichards** [comment]
I wasn't suggesting istypes, because yes, that's too much.
I was thinking some kind of bridge procs, but if ```add_overlay()``` does it, that's fine enough.

**Pull request merged**